modules/build/src/main/scala/scala/build/CLangInstaller.scala

```diff
@@ -0,0 +1,48 @@
+package scala.build
+
+import os.Path
+
+import scala.build.errors.BuildException
+import scala.build.internal.Runner
+import scala.util.Properties
+
+final class CLangInstallException extends BuildException("Failed to Install CLang")
+
+object CLangInstaller {
+  /*
+  Installer based on micromamba attempts to install mamba and llvm based on that
+  Following https://gist.github.com/wolfv/fe1ea521979973ab1d016d95a589dcde
+   */
+  val mambaApiUrl = "https://micromamba.snakepit.net/api/micromamba"
+  def mambaBinaryUrl = Properties.osName match {
+    case os if os startsWith "Windows"  => s"$mambaApiUrl/win-64/latest"
+    case os if os startsWith "Mac OS X" => s"$mambaApiUrl/osx-64/latest"
+    case os if os startsWith "Linux"    => s"$mambaApiUrl/linux-64/latest"
+  }
+
+  def install(microMambaArchive: Path, logger: Logger): Either[BuildException, Path] =
+    install(microMambaArchive, os.Path("/tmp/clang_output"), logger = logger)
+
+  def install(
+    microMambaArchive: Path,
+    outputPath: Path,
+    logger: Logger
+  ): Either[BuildException, Path] =
+    TarArchive.decompress(os.read.inputStream(microMambaArchive), outputPath)
```

Comment:
If you need to uncompress an archive, you can use `coursier.cache.ArchiveCache`. One is created [here](https://github.com/VirtusLab/scala-cli/blob/f999de23f6b1aaead9078b0b9d1ae0cde55c3c84/modules/build/src/main/scala/scala/build/options/BuildOptions.scala#L268) for example. Then its `get` method will give you a directory where the archive is extracted.

Reply:
nice 👍